File: server/datasets/python_source-code/Multiple-E4/type3_while_1_substring_search.py
def strStr(haystack, needle):
    if not needle: return 0
    
    i = 0
    while i <= len(haystack) - len(needle):
        for j in range(len(needle)):
            if haystack[i + j] != needle[j]:
                break
            if j == len(needle) - 1:
                return i
        i += 1
        
    return -1

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_e4/type3_while_1_substring_search.java
public class Solution {
    public int strStr(String haystack, String needle) {
        if (needle == null || needle.length() == 0) return 0;
        
        int i = 0;
        while (i <= haystack.length() - needle.length()) {
            for (int j = 0; j < needle.length(); j++) {
                if (haystack.charAt(i + j) != needle.charAt(j)) {
                    break;
                }
                if (j == needle.length() - 1) {
                    return i;
                }
            }
            i++;
        }
        return -1;
    }
}
